### PR TITLE
feat: As a Purchaser, I want to compare different supplier quotations, so I can choose the best deal

### PR DESCRIPTION
**Pull request by mymi14s**
## Supplier Quotation Comparison Automation
This feature automatically filter and select quotation based on four(4) analysis options:

1. Best rate from one supplier (focuses on supplier quotation grand_total)
2. Best rate from many supplier (item rate)
3. Earliest delivery date
4. Custom (some items[rate] others[earliest delivery date) 

## Analysis, design  and solution
The problem space deals with quotations from multiple suppliers for many items where a purchaser manually filters and choose the best rate/delivery date based on use-case.
The system takes into consideration two case scenario:

1. Supplier have all request items and quantity.
2. Some items and quantities (not available).

The first scenario will be a priority but if not met, the second scenario will be considered.
Once comparison is complete, a button will be available to create Purchase Order with consideration of selected items and suppliers. One or more PO(s) will be created based on number of selected suppliers.

## Output screenshots (optional)
image

## Areas affected and ensured
The comparison business logic is written entirely in client side(javascript)

### CODE REVIEW

File: one_fm/purchase/doctype/quotation_comparison_sheet/quotation_comparison_sheet.js [hunk 1]
```diff
@@ -139,3 +213,353 @@ var set_quotation_item_details = function(frm, item, quotation) {
...
+		}, 'Analyse');
+
+		// best_price_many_suppliers
+		frm.add_custom_button('Earliest Delivery', () => {
+			earliest_delivery(frm);
+		}, 'Analyse');
+
+		// best_price_many_suppliers
+		frm.add_custom_button('Custom', () => {
+			customer_filter(frm);
```
Comment (pjamsheer) [inline one_fm/purchase/doctype/quotation_comparison_sheet/quotation_comparison_sheet.js]: customer_filter. I hope you set it for custom_filter

File: one_fm/purchase/doctype/quotation_comparison_sheet/quotation_comparison_sheet.js [hunk 2]
```diff
@@ -139,3 +213,353 @@ var set_quotation_item_details = function(frm, item, quotation) {
...
+	let best_quotation_items = frm.doc.quotation_items.filter(
+		item => item.quotation === ordered_quotations.quotation
+	);
+	// append to selected filtered table
+	complete_filters_table(frm, best_quotation_items, 'Earliest Delivery');
+}
+
+
+// custom filter
+let customer_filter = (frm)=>{
```
Comment (pjamsheer) [inline one_fm/purchase/doctype/quotation_comparison_sheet/quotation_comparison_sheet.js]: customer_filter

File: one_fm/purchase/doctype/quotation_comparison_sheet/quotation_comparison_sheet.js [hunk 3]
```diff
@@ -139,3 +213,353 @@ var set_quotation_item_details = function(frm, item, quotation) {
...
+				// {
+				// 	fieldtype: '', label: __('Company'), default: frappe.defaults.get_default('company'),
+				// 	options: 'Company', fieldname: 'company', reqd: 1
+				// },
+				// {
+				// 	fieldtype: 'Link', label: __('POS Profile'),
+				// 	options: 'POS Profile', fieldname: 'pos_profile', reqd: 1,
+				// 	get_query: () => pos_profile_query,
+				// 	onchange: () => fetch_pos_payment_methods()
+				// },
```
Comment (pjamsheer) [inline one_fm/purchase/doctype/quotation_comparison_sheet/quotation_comparison_sheet.js]: Remove unused codes

**Comment by pjamsheer** [comment]
@mymi14s Great work ! There are some minor suggestions from my side !

File: one_fm/purchase/doctype/quotation_comparison_sheet/quotation_comparison_sheet.js (hunk 1 above)
Comment (mymi14s) [reply one_fm/purchase/doctype/quotation_comparison_sheet/quotation_comparison_sheet.js]: Ah, a rename will be required

File: one_fm/purchase/doctype/quotation_comparison_sheet/quotation_comparison_sheet.js (hunk 3 above)
Comment (mymi14s) [reply one_fm/purchase/doctype/quotation_comparison_sheet/quotation_comparison_sheet.js]: Unused code removed and customer_filter renamed to custom filter.

**Review by pjamsheer — APPROVED** [review]

**Pull request merged**